dquartic/dia-nn to massdash feature extraction/Feature_extraction.py

```
@@ -0,0 +1,397 @@
+                    ms2_gpu = ms2_gpu.rename(columns={'rt': 'RetentionTime',
+                                                      'mz': 'Fragment_mz',
+                                                      'precursor_mz': 'Precursor_mz',
+                                                      'int': 'Intensity'})
+                    ms2_list.append(ms2_gpu.to_pandas())
+            
+            # Save current batch results as CSV files
+            if ms1_list:
+                batch_ms1_df = pd.concat(ms1_list, ignore_index=True)
+                batch_ms1_filename = os.path.join(self.ms1_dir, f"ms1_features_batch_{batch_idx}.csv")
+                batch_ms1_df.to_csv(batch_ms1_filename, index=False)
+                ms1_all_batches.append(batch_ms1_df)
+            if ms2_list:
+                batch_ms2_df = pd.concat(ms2_list, ignore_index=True)
+                batch_ms2_filename = os.path.join(self.ms2_dir, f"ms2_features_batch_{batch_idx}.csv")
+                batch_ms2_df.to_csv(batch_ms2_filename, index=False)
+                ms2_all_batches.append(batch_ms2_df)
+            
+            batch_elapsed = time.time() - batch_start_time
+            logging.info(f"Finished processing batch {batch_idx+1} in {batch_elapsed:.2f} seconds.")
+        
+        # Combine all batch results and export final combined files.
+        if ms1_all_batches:
+            final_ms1_df = pd.concat(ms1_all_batches, ignore_index=True)
+            final_ms1_file = os.path.join(self.ms1_dir, "ms1_features_all_peptides_combined.csv")
+            final_ms1_df.to_csv(final_ms1_file, index=False)
+        else:
+            final_ms1_file = None
+        if ms2_all_batches:
+            final_ms2_df = pd.concat(ms2_all_batches, ignore_index=True)
+            final_ms2_file = os.path.join(self.ms2_dir, "ms2_features_all_peptides_combined.csv")
+            final_ms2_df.to_csv(final_ms2_file, index=False)
+        else:
+            final_ms2_file = None
```

If ms1_all_batches and ms2_all_batches are empty, should we throw an error or throw a logging error to say that there is no data being written? Which likely means something is wrong, or none of the peptides from the DIA-NN report were found with MassDash's extraction?

---

Agreed, this is a good way to handle potential issues. I've modified the code to raise an error if empty files are encountered.